src/main/java/com/idrsolutions/microservice/JPedalServlet.java

```diff
@@ -270,88 +273,99 @@ protected boolean validateRequest(final HttpServletRequest request, final HttpSe
         return true;
     }
 
-    private static void convertPDF(final Mode mode, final String userPdfFilePath, final String outputDirStr,
-                                   final String fileNameWithoutExt, final Map<String, String> paramMap, final ErrorTracker durationTracker) throws Exception {
+    private ProcessUtils.Result convertFile(final Map<String, String> conversionParams,
+        final String uuid, final String webappDirectory, final File inputPdf,
+        final File outputDir, final long maxDuration) {
+
+
+        final ArrayList<String> commandArgs = new ArrayList<>();
+        commandArgs.add("java");
+
+        final Properties properties = (Properties) getServletContext().getAttribute(BaseServletContextListener.KEY_PROPERTIES);
+        final int memoryLimit = Integer.parseInt(properties.getProperty(BaseServletContextListener.KEY_PROPERTY_CONVERSION_MEMORY));
+
+        final int remoteTrackerPort = Integer.parseInt((String) properties.get(BaseServletContextListener.KEY_PROPERTY_REMOTE_TRACKING_PORT));
 
+        float scaling = (conversionParams.containsKey("scaling") ? Float.parseFloat(conversionParams.get("scaling")) * 1.52f : 1.52f);
+        if (memoryLimit > 0) {
+            commandArgs.add("-Xmx" + memoryLimit + "M");
+        }
+
+        //Set settings
+        commandArgs.add("-Dcom.idrsolutions.remoteTracker.port=" + remoteTrackerPort);
+        commandArgs.add("-Dcom.idrsolutions.remoteTracker.uuid=" + uuid);
+
+        //Add jar and input / output
+        commandArgs.add("-cp");
+        commandArgs.add(webappDirectory);
+
+        final Mode mode = Mode.valueOf((conversionParams.remove("mode")));
         switch (mode) {
             case convertToImages:
-                ConvertPagesToImages.writeAllPagesAsImagesToDir(
-                        userPdfFilePath,
-                        outputDirStr + fileSeparator + fileNameWithoutExt + fileSeparator,
-                        paramMap.get("format"),
-                        Float.parseFloat(paramMap.getOrDefault("scaling", "1.0")),
-                        paramMap.get("password"),
-                        durationTracker);
+                commandArgs.add("org.jpedal.examples.images.ConvertPagesToImages");
+                commandArgs.add(inputPdf.getAbsolutePath());
+                commandArgs.add(outputDir.getAbsolutePath());
+                commandArgs.add(conversionParams.get("format"));
+                commandArgs.add(String.valueOf(scaling));
                 break;
-                case extractImages:{
-                final String type = paramMap.get("type");
+            case extractImages:
+                String type = conversionParams.get("type");
```

Comment:
Thoughts on renaming the variable so it can be final?
```suggestion
                final String imageType = conversionParams.get("type");
```